src/ray/raylet/node_manager.cc

```diff
@@ -360,8 +373,28 @@ void NodeManager::ClientAdded(const ClientTableData &client_data) {
   RAY_LOG(DEBUG) << "[ClientAdded] Received callback from client id " << client_id;
   if (client_id == gcs_client_->client_table().GetLocalClientId()) {
     // We got a notification for ourselves, so we are connected to the GCS now.
-    // Save this NodeManager's resource information in the cluster resource map.
-    cluster_resource_map_[client_id] = initial_config_.resource_config;
+
+    // Delay execution to make sure,
+    // 1) Client cache is up to date.
+    // 2) Lookup resource table after subscribe.
+    io_service_.post([this]() {
```

Comment:
Perhaps I'm missing something, but instead of using this lambda, why can't we update the resources of the remote clients at [line 412](https://github.com/ray-project/ray/blob/411f22b53b3000bd6a863cac4a2aa42da4a5e9fe/src/ray/raylet/node_manager.cc#L412) when they're added to the local raylet? Even if client cache is not up to date, subsequent notifications should handle update/delete in the local raylet.

Reply:
After discussed with @raulchen, we think that it's OK to put the lookup at the end of `ClientAdded` and only lookup resources for one client. So I updated the code as you suggested to avoid the lambda.